src/EditorFeatures/Core/Implementation/LineSeparators/AdornmentManager.cs

```diff
@@ -302,6 +312,33 @@ private void UpdateInvalidSpans()
             }
         }
 
+        private NormalizedSpanCollection GetCollapsedSpans()
+        {
+            var elisionSnapshot = _textView.VisualSnapshot as IProjectionSnapshot;
+            if (elisionSnapshot == null)
+            {
+                return NormalizedSpanCollection.Empty;
+            }
+
+            // Get the source spans from the visual snapshot.  The source spans will be for the 
+            // uncollapsed regions.  So we can infer the collapsed regions for the spans between
+            // the uncollapsed ones.
+
+            var sourceSpans = elisionSnapshot.GetSourceSpans();
```

Comment:
This list could be very big. Instead of collecting all of it, why not call IProjectionBuffer.MapFromSourceSnapshot on the end point and see if it maps or not? Each call should be O(log n) and you'll only do number of calls of actual lines versus collecting lots of data up front.